src/Equinor.Procosys.Library.Query/GetAllDisciplines/GetAllDisciplinesQueryHandler.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Equinor.Procosys.Library.Query.Client;
+using MediatR;
+using Microsoft.Extensions.Options;
+using ServiceResult;
+
+namespace Equinor.Procosys.Library.Query.GetAllDisciplines
+{
+    public class GetAllDisciplinesQueryHandler : IRequestHandler<GetAllDisciplinesQuery, Result<IEnumerable<DisciplineDto>>>
+    {
+        private readonly string _apiVersion;
+        private readonly Uri _baseAddress;
+        private readonly IBearerTokenApiClient _mainApiClient;
+
+        public GetAllDisciplinesQueryHandler(
+            IBearerTokenApiClient mainApiClient,
+            IOptionsMonitor<MainApiOptions> options)
+        {
+            _apiVersion = options.CurrentValue.ApiVersion;
+            _baseAddress = new Uri(options.CurrentValue.BaseAddress);
+            _mainApiClient = mainApiClient;
+        }
+
+        public async Task<Result<IEnumerable<DisciplineDto>>> Handle(GetAllDisciplinesQuery request, CancellationToken cancellationToken)
+        {
+            var url = $"{ _baseAddress}Library/Disciplines" +
+                $"?plantId={request.Plant}" +
+                string.Join("", request.Classifications
```

Comment:
Add Classification filter only when any given

Reply:
It won't be included if there are no items in the list.

Reply:
ok, see comment in GetAllDisciplinesQueryHandlerTests.cs regarding testing if no classifications